include/nbl/video/utilities/IGPUObjectFromAssetConverter.h

```diff
@@ -120,24 +121,61 @@ class IGPUObjectFromAssetConverter
             // @sadiuk put here more parameters if needed
 
             SPerQueue perQueue[EQU_COUNT];
+            
+            inline void beginCommandBuffer(E_QUEUE_USAGE queueUsage)
+            {
+                uint32_t queue = static_cast<uint32_t>(queueUsage);
+                assert(perQueue[queue].cmdbuf);
+                perQueue[queue].cmdbuf->begin(IGPUCommandBuffer::EU_ONE_TIME_SUBMIT_BIT);
+            }
 
-
+            inline void beginCommandBuffers()
+            {
+                for (auto i=0; i<EQU_COUNT; i++)
+                {
+                    beginCommandBuffer(static_cast<E_QUEUE_USAGE>(i));
+                }
+            }
+            
+            inline void endUnusedCommandBuffers()
+            {
+                for(auto & queue : perQueue)
+                {
+                    if(queue.cmdbuf->getState() == IGPUCommandBuffer::ES_RECORDING)
+                        queue.cmdbuf->end();
+                }
+            }
+            
             //
-            inline void waitForCreationToComplete()
+            inline void waitForCreationToComplete(bool resetFencesAfterWait = false)
             {
-                IGPUFence* fences[EQU_COUNT];
+                IGPUFence* fence_ptrs[EQU_COUNT];
                 uint32_t count = 0;
                 for (auto i=0; i<EQU_COUNT; i++)
                 {
-                    auto pFence = perQueue[i].fence;
-                    if (pFence && pFence->get()) // user wanted, and something actually got submitted
-                        fences[count++] = pFence->get();
+                    auto pFence = fences[i];
+                    if (pFence && pFence.get()) // user wanted, and something actually got submitted
+                        fence_ptrs[count++] = pFence.get();
+                }
+                device->blockForFences(count,fence_ptrs);
+                if(resetFencesAfterWait)
+                {
+                    device->resetFences(count, fence_ptrs);
                 }
-                device->blockForFences(count,fences);
+                setFencesToNull();
+                endUnusedCommandBuffers();
```

Comment:
no, you must end before you submit

and wait for fences after you submit

Reply:
The user (probably CommonAPI) fills cmdBufs per Queue in the CPU2GPU Params, if the user calls beginCommandBuffers but don't get used (not going through Recording->Exe->Pending->Invalid->Initial) 
Then it will remain in the Recording state. 
I wanted to solve that by ending the command buffer but now this seems dumb and I should probably reset all the commandBuffers after waiting for the fences.
I should get some sleep now 💤

Reply:
> I should probably reset all the commandBuffers after waiting for the fences.
Very good idea

Also remember, begin!=reset so they'll be back in initial state ;)

Reply:
done